File: daq/slc/dqm/src/Graph1D.cc
#include "daq/slc/dqm/Graph1D.h"

#include "daq/slc/dqm/DoubleArray.h"

using namespace Belle2;

Graph1D::Graph1D() : Graph1()
{
  m_data = new DoubleArray(0);
}

Graph1D::Graph1D(const Graph1D& h) : Graph1(h)
{
  m_data = new DoubleArray(*h.m_data);
}

Graph1D::Graph1D(const std::string& name, const std::string& title,
                 int nbinx, double xmin, double xmax, double ymin, double ymax)
  : Graph1(name, title, nbinx, xmin, xmax, ymin, ymax)
{
  m_data = new DoubleArray(nbinx * 2);
}

Graph1D::~Graph1D() throw() {}

std::string Graph1D::getDataType() const throw()
{
  return std::string("g1D");
}

File: daq/slc/extra/javalib/css/org.csstudio.belle2.daq/src/org/belle2/daq/dqm/Graph1D.java
package org.belle2.daq.dqm;


public class Graph1D extends Graph1 {

	public Graph1D() {
		super();
		data = new DoubleArray(0);
	}
	public Graph1D(String name, String title, int nbinx, double xmin, double xmax,
			double ymin, double ymax) {
		super(name, title, nbinx, xmin, xmax, ymin, ymax);
		data = new DoubleArray(nbinx*2);
	}
	public Graph1D(Graph1D h) {
		super();
		data = new DoubleArray(0);
	}
	public String getDataType() {
		return "g1D";
	}

}
